activation/atxdb.go

```diff
@@ -228,6 +218,43 @@ func (db *ActivationDb) ValidateAtx(atx *types.ActivationTx) error {
 	return nil
 }
 
+// ContextuallyValidateAtx ensures that the previous ATX referenced is the last known ATX for the referenced miner ID.
+// If a previous ATX is not referenced, it validates that indeed there's no previous known ATX for that miner ID.
+func (db *ActivationDb) ContextuallyValidateAtx(atx *types.ActivationTx) error {
+	if atx.PrevATXId != *types.EmptyAtxId {
+		prevAtxIds, err := db.GetNodeAtxIds(atx.NodeId)
```

Comment:
I think that you can put this outside the if brackets, as both the "if" and "else" call it...

Reply:
yes, but the error handling is different and I don't want to split the code from the error handling.